crates/biome_js_semantic/src/semantic_model/model.rs

```diff
@@ -331,52 +327,107 @@ impl SemanticModel {
 
     pub fn as_binding(&self, binding: &impl IsBindingAstNode) -> Binding {
```

Comment:
I don't think this API should exist anymore, mostly because we lose information and the function is not "clear" enough. What kind of binding are we retrieving? When requesting a binding, we don't know if it's a `Value` or a `Type`